store-api-server/src/payment/service/payment.service.ts

```diff
@@ -1590,4 +1656,19 @@ WHERE country_short = $1
       +ipParts[3]
     );
   }
+
+  getCurrency(provider: PaymentProvider, currency: string): string {
+    switch (provider) {
+      case PaymentProvider.TEZPAY:
+        return 'XTZ';
+      case PaymentProvider.WERT:
+        return 'USD';
+      case PaymentProvider.SIMPLEX:
+        return 'USD';
+      case PaymentProvider.STRIPE:
+        return currency === PaymentProvider.TEZPAY ? 'USD' : currency;
```

Comment:
This should be `currency === 'XTZ' ? (BASE_CURRENCY === 'XTZ' ? 'USD' : BASE_CURRENCY) : currency`

Reply:
Should be solved now